playbook/initializeDomain.ts

```diff
@@ -20,35 +25,94 @@ task('initializeDomain', 'Initialize domain name and activate it')
       const multipassDeployment = await deployments.get('Multipass');
       registrarAddress = registrarAddress ?? registrar;
       const multipassContract = new hre.ethers.Contract(
         multipassDeployment.address,
         multipassDeployment.abi,
         hre.ethers.provider.getSigner(owner),
       ) as MultipassDiamond;
-      const tx = await multipassContract.initializeDomain(
+      /*const tx = await multipassContract.initializeDomain(
         registrarAddress,
         freeRegistrationsNumber,
         hre.ethers.utils.parseEther(fee),
         hre.ethers.utils.formatBytes32String(domain),
         hre.ethers.utils.parseEther(reward),
         hre.ethers.utils.parseEther(discount),
       );
-      console.log(tx.wait(1));
+      console.log(tx.wait(1));*/
 
       if(activate === true) {
-        const tx = await multipassContract
+        /*const tx = await multipassContract
           .activateDomain(hre.ethers.utils.formatBytes32String(domain));
         console.log(tx.wait(1));
-        console.log('Domain name "' + domain + '" successfully initialized and activated!')
+        console.log('Domain name "' + domain + '" successfully initialized and activated!')*/
+      }
+
+      if(username) {
+        const playerId = crypto.randomUUID().slice(0, 31);
+        const registrarMessage = {
+          name: hre.ethers.utils.formatBytes32String(username),
+          id: hre.ethers.utils.formatBytes32String(playerId),
+          domainName: hre.ethers.utils.formatBytes32String(domain),
+          deadline: hre.ethers.BigNumber.from(9999),
+          nonce: hre.ethers.BigNumber.from(0),
+        };
+        let signer =  await hre.ethers.getSigner(player);
+    
+        const validSignature = await signRegistrarMessage(registrarMessage, multipassDeployment.address, signer, hre);
+
+        let applicantData: LibMultipass.RecordStruct = {
+          name: hre.ethers.utils.formatBytes32String(username),
+          id: hre.ethers.utils.formatBytes32String(playerId),
+          wallet: player,
+          nonce: 0,
+          domainName: hre.ethers.utils.formatBytes32String(domain),
+        };
+
+        const emptyUserQuery: LibMultipass.NameQueryStruct = {
+          name: hre.ethers.utils.formatBytes32String(''),
+          id: hre.ethers.utils.formatBytes32String(''),
+          domainName: hre.ethers.utils.formatBytes32String(''),
+          wallet: hre.ethers.constants.AddressZero,
+          targetDomain: hre.ethers.utils.formatBytes32String(''),
+        };
+
+        const tx = await multipassContract
+            .connect(player)
+            .register(
+              applicantData,
+              registrarMessage.domainName,
+              validSignature,
+              registrarMessage.deadline,
+              emptyUserQuery,
+              hre.ethers.constants.HashZero,
+            )
+         console.log(tx);   
       }
     },
   );
 
+  export const signRegistrarMessage = async (
```

Comment:
I would suggest just instantiate locally a client and use it, why would you need to call a function (and export it)?